pkg/agent/monitortool/latency_store_test.go

```diff
@@ -0,0 +1,187 @@
+		{
+			key:           "10.244.2.2",
+			expectedEntry: NodeIPLatencyEntry{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"Node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			entry, _ := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.expectedEntry.LastMeasuredRTT, entry.LastMeasuredRTT)
+			assert.Equal(t, tt.expectedEntry.LastSendTime, entry.LastSendTime)
+			assert.Equal(t, tt.expectedEntry.LastRecvTime, entry.LastRecvTime)
+		})
+	}
+}
+
+func TestLatencyStore_SetNodeIPLatencyEntry(t *testing.T) {
+	tests := []struct {
+		key           string
+		updatedEntry  NodeIPLatencyEntry
+		expectedEntry NodeIPLatencyEntry
+	}{
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry,
+			expectedEntry: entry,
+		},
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry2,
+			expectedEntry: entry2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"Node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			mutator := func(entry *NodeIPLatencyEntry) {
+				entry.LastSendTime = tt.updatedEntry.LastSendTime
+				entry.LastRecvTime = tt.updatedEntry.LastRecvTime
+				entry.LastMeasuredRTT = tt.updatedEntry.LastMeasuredRTT
+			}
+			latencyStore.SetNodeIPLatencyEntry(tt.key, mutator)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.expectedEntry, entry)
+			assert.True(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_DeleteStaleNodeIPs(t *testing.T) {
+	testKey := "10.244.2.1"
+
+	latencyStore := &LatencyStore{
+		isNetworkPolicyOnly: false,
+		nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+			"10.244.2.1": &entry,
+		},
+		nodeTargetIPsMap: map[string][]net.IP{
+			"Node1": {net.ParseIP("10.244.2.1")},
+		},
+	}
+
+	// Remove Node
+	latencyStore.deleteNode(&corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Node1",
+		},
+	})
+
+	// Check if the entry is deleted
+	_, ok := latencyStore.getNodeIPLatencyEntry(testKey)
+	assert.True(t, ok)
+
+	// Send the echo from the node
+	mutator := func(entry *NodeIPLatencyEntry) {}
+	latencyStore.SetNodeIPLatencyEntry(testKey, mutator)
+	_, ok = latencyStore.getNodeIPLatencyEntry(testKey)
+	assert.True(t, ok)
+
+	// Check if the entry is deleted
+	latencyStore.DeleteStaleNodeIPs()
+	_, ok = latencyStore.getNodeIPLatencyEntry(testKey)
+	assert.False(t, ok)
+
+	// Receive the reply from the node
+	mutator = func(entry *NodeIPLatencyEntry) {}
+	latencyStore.SetNodeIPLatencyEntry(testKey, mutator)
+	_, ok = latencyStore.getNodeIPLatencyEntry(testKey)
+	assert.True(t, ok)
```

Comment:
remove as this is not relevant to the test

Reply:
OK.